File: LMIS/LMIS.Web/Modules/ReaderManage/UserBill/UserBillColumns.cs
using LMIS.Modules.ReaderManage.UserBill;
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.ReaderManage.Columns;

[ColumnsScript("ReaderManage.UserBill")]
[BasedOnRow(typeof(UserBillRow), CheckNames = true)]
public class UserBillColumns
{
    //[AlignRight]
    //public long UserBillId { get; set; }
    public string UserName { get; set; }
    [Width(100)]
    public BillTypeEnum Type { get; set; }
    public decimal Fee { get; set; }
    [Width(300)]
    public string Remark { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime UpdateTime { get; set; }
}

File: LMIS/LMIS.Web/Modules/ServerTypes/ReaderManage/UserBillColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { BillTypeEnum } from "../Modules/ReaderManage.UserBill.BillTypeEnum";
import { UserBillRow } from "./UserBillRow";

export interface UserBillColumns {
    UserName: Column<UserBillRow>;
    Type: Column<UserBillRow>;
    Fee: Column<UserBillRow>;
    Remark: Column<UserBillRow>;
    UpdateTime: Column<UserBillRow>;
}

export class UserBillColumns extends ColumnsBase<UserBillRow> {
    static readonly columnsKey = 'ReaderManage.UserBill';
    static readonly Fields = fieldsProxy<UserBillColumns>();
}

[BillTypeEnum]; // referenced types